Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -664,6 +664,24 @@ struct ParametersAndReturnValidator {
 
 // MARK: Helper extensions
 
+private extension Return {
+    /// Applies a basic heuristic to give an indication if the return value documentation possibly documents what happens when an error occurs
+    func possiblyDocumentsFailureBehavior() -> Bool {
+        contents.contains(where: { markup in
+            let formatted = markup.format().lowercased()
+            
+            // Check if the authored markup contains one of a handful of words as an indication that it possibly documents what happens when an error occurs.
+            return returnValueDescribesErrorRegex.firstMatch(in: formatted, range: NSRange(formatted.startIndex ..< formatted.endIndex, in: formatted)) != nil
+        })
+    }
+}
+    
+/// A regular expression that finds the words; "error", "fail", "fails", "failure", "failures", "nil", and "null".
+/// These words only match at word boundaries or when surrounded by single backticks ("`").
+///
+/// This is used as a heuristic to give an indication if the return value documentation possibly documents what happens when an error occurs.
+private let returnValueDescribesErrorRegex = try! NSRegularExpression(pattern: "(\\b|`)(error|fail(ure)?s?|nil|null)(\\b|`)", options: .caseInsensitive)
```

Comment:
Can we use [Regex](https://developer.apple.com/documentation/swift/regex) instead? It has an option for case insensitivity as well [1].

[1]: https://developer.apple.com/documentation/swift/regex/ignorescase(_:)

Reply:
`Regex` is available as of macOS 13 and we currently support macOS 12 (bumped very recently). 

We probably *could* bump to macOS 13 but I didn't do that for this fix because this regular expression wasn't all that complicated.